crates/next-api/src/server_actions.rs

```diff
@@ -510,16 +382,41 @@ impl AllActions {
 
 /// Maps the hashed action id to the action's exported function name.
 #[turbo_tasks::value(transparent)]
-struct ActionMap(FxIndexMap<String, String>);
+pub struct ActionMap(FxIndexMap<String, String>);
 
 /// An Option wrapper around [ActionMap].
 #[turbo_tasks::value(transparent)]
 struct OptionActionMap(Option<ResolvedVc<ActionMap>>);
 
-#[turbo_tasks::value_impl]
-impl OptionActionMap {
-    #[turbo_tasks::function]
-    pub fn none() -> Vc<Self> {
-        Vc::cell(None)
-    }
+type LayerAndActions = (ActionLayer, ResolvedVc<ActionMap>);
+/// A mapping of every module module containing Server Actions, mapping to its layer and actions.
+#[turbo_tasks::value(transparent)]
+pub struct AllModuleActions(HashMap<ResolvedVc<Box<dyn Module>>, LayerAndActions>);
+
+#[turbo_tasks::function]
+pub async fn map_server_actions(graph: Vc<SingleModuleGraph>) -> Result<Vc<AllModuleActions>> {
+    let actions = graph
+        .await?
+        .enumerate_nodes()
```

Comment:
enumerate_nodes is a bit inefficient here. It could use `iter_nodes()` which could use `DiGraph:: node_weights`